pytest-django: tests fail due to sandbox on aarch64-darwin

### Describe the bug

`pytest-django` fails to build on aarch64-darwin due to failing tests due to a sandbox issue.

See also: https://github.com/NixOS/nixpkgs/pull/282912

### Steps To Reproduce

```console
$ nix build --rebuild .#python311.pkgs.pytest-django
error: builder for '/nix/store/l35qps1mk6nxhwwa9ijblsk6xw8q2fnf-python3.11-pytest-django-4.7.0.drv' failed with exit code 1;
       last 10 log lines:
       > ERROR tests/test_fixtures.py::TestLiveServer::test_change_settings - socket.gaierror: [Errno 8] nodename nor servname provided, or not known
       > ERROR tests/test_fixtures.py::TestLiveServer::test_transactions - socket.gaierror: [Errno 8] nodename nor servname provided, or not known
       > ERROR tests/test_fixtures.py::TestLiveServer::test_serve_static_dj17_without_staticfiles_app - socket.gaierror: [Errno 8] nodename nor servname provided, or not known
       > ERROR tests/test_fixtures.py::TestLiveServer::test_db_changes_visibility - socket.gaierror: [Errno 8] nodename nor servname provided, or not known
       > ERROR tests/test_fixtures.py::TestLiveServer::test_item - socket.gaierror: [Errno 8] nodename nor servname provided, or not known
       > ERROR tests/test_fixtures.py::TestLiveServer::test_url - socket.gaierror: [Errno 8] nodename nor servname provided, or not known
       > FAILED tests/test_fixtures.py::TestLiveServer::test_specified_port_django_111 - PermissionError: [Errno 1] Operation not permitted
       > FAILED tests/test_fixtures.py::TestLiveServer::test_serve_static_with_staticfiles_app - Failed: nomatch: '*test_a*PASSED*'
       > ============= 2 failed, 202 passed, 2 skipped, 10 errors in 25.24s =============
       > /nix/store/ymqnr6q4slam0ns76mjpcjry70fa8h3i-stdenv-darwin/setup: line 1591: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/l35qps1mk6nxhwwa9ijblsk6xw8q2fnf-python3.11-pytest-django-4.7.0.drv'.
$ nix build --rebuild .#python311.pkgs.pytest-django --option sandbox false
$ echo $?
0
```

### Expected behavior

Tests should pass, allowing the package build to succeed.

### Notify maintainers

none listed for pytest-django specifically, @hexa for django in general

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2.1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.2`
 - channels(n8henrie): `""`
 - channels(root): `""`
 - nixpkgs: `/nix/store/64npyyxn3bb6cxciz12yigldfc2p9fh1-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pytest-django/default.nix
+++ b/pkgs/development/python-modules/pytest-django/default.nix
@@ -51,6 +51,8 @@ buildPythonPackage rec {
     "test_settings_restored"
   ];
 
+  __darwinAllowLocalNetworking = true;
+
   meta = with lib; {
     description = "py.test plugin for testing of Django applications";
     homepage = "https://pytest-django.readthedocs.org/en/latest/";

